Jetcaster/core/src/main/java/com/example/jetcaster/core/data/repository/CategoryStore.kt

```diff
@@ -26,32 +26,67 @@ import com.example.jetcaster.core.data.database.model.PodcastCategoryEntry
 import com.example.jetcaster.core.data.database.model.PodcastWithExtraInfo
 import kotlinx.coroutines.flow.Flow
 
+interface CategoryStore {
```

Comment:
I would say you don't need to have this abstraction, because this abstraction is provided by room by having the DAO abstract classes. Basically what you can have is implement `TestCategoriesDao` (and others) and have the simpler logic there. While the DAO class is annotated, from the perspective of the tests it's just a pure abstract class.

Reply:
I originally went with this approach but then decided to create an interface for the `*Store` objects because creating `Test*Dao` objects was too time-consuming :/ This would probably be the right approach though if I were writing tests for all layers.